W3ChampionsStatisticService/Matches/MatchRepository.cs

```diff
@@ -176,12 +175,24 @@ public async Task<int> GetFloIdFromId(string gameId)
         return (match == null || match.FloMatchId == null) ? 0 : match.FloMatchId.Value;
     }
 
-    public Task<long> Count(
-        int season,
-        GameMode gameMode)
+    public Task<long> Count(int season, GameMode gameMode, HeroType hero = HeroType.AllFilter)
+    {
+        var filter = GetLoadFilter(season, gameMode, hero);
+        return CreateCollection<Matchup>().CountDocumentsAsync(filter);
+    }
+
+    private FilterDefinition<Matchup> GetLoadFilter(int season, GameMode gameMode, HeroType hero = HeroType.AllFilter)
     {
-        return CreateCollection<Matchup>().CountDocumentsAsync(m =>
-                gameMode == m.GameMode && m.Season == season);
+        var builder = Builders<Matchup>.Filter;
+        var filter = builder.Eq(m => m.GameMode, gameMode) & builder.Eq(m => m.Season, season);
+
+        if (hero != HeroType.AllFilter && hero != HeroType.Unknown)
+        {
+            var heroFilter = builder.Where(m => m.Teams.Any(t => t.Players.Any(p => p.Heroes.Count > 0 && p.Heroes[0].Id == hero)));
```

Comment:
How come you're only checking Heroes[0]?

Reply:
Suggestion from @francislavoie to only filter on the first hero instead of all heroes

Reply:
Hmm okay, feels odd that we're only selecting the first hero, but well

Reply:
Well it's more about "what's the first hero in the build order". Getting MK first is not the same thing as being it second or third, that's a different build entirely, so filtering on games with that too wouldn't do much. Think about it, a human player might alternate between 2 or 3 different strategies, but all of the strats could get the same one hero eventually (except for fast games of course that end with one hero).